### ISSUE TITLE
Clean up interface to graph coloring algorithms

### ISSUE THREAD

**Issue by lucbv** [kokkos/kokkos-kernels#242]
Currently two similar yet different calls can be made to obtain distance 2 coloring: `d2_graph_color` and `graph_color_d2`.
This is a little confusing and it also makes it hard for an application to use the correct algorithms. I think that one of these two function should be deprecated and replaced by an overload of the other one if necessary?

**Comment by aprokop** [comment]
Do different enum names call different functions, or the same?

**Comment by lucbv** [comment]
I am not sure since I did not look into the details of what these two functions are doing but yes if we could simplify things using the `enum` that already exist to switch between algorithms that would be best.

**Comment by ndellingwood** [comment]
@william76 does your PR #375 address this issue or does this issue require additional work from your PR?

**Comment by william76** [comment]
@ndellingwood 
Yeah, this thing is super confusing.  The _old_ D2 Coloring is in `src/graph/KokkosGraph_GraphColor.hpp` (this file is changed in this PR but I have the `src/graph/KokkosGraph_graph_color.hpp` wrapper header for backwards compatibility.  For backwards compatibility, I've kept the `d2_graph_color() function there which has access to the `COLORING_SERIAL2` variant, which is the serial one that Mehmet implemented.

The new Distance-2 interface, which is in `src/graph/KokkosGraph_Distance2Color.hpp` uses `graph_compute_distance2_color()` to compute the distance-2 coloring.  This one can can access the various D2 coloring algorithms.

I'd like to get all the D2 coloring stuff out of the GraphColorHandle and into GraphColorDistance2Handle but since the serial d2 is used in SPGEMM (and possibly other places outside of KokkosKernels) simply removing that is a little tricky. 

The naming still has that confusion there and I'd like to get rid of the `d2_graph_color()` function all together.  We should schedule this one to go away since you can get the Serial D2 now from the new stuff, how do you guys typically handle adding in deprecation warnings, etc.?

**Comment by brian-kelley** [comment]
This was taken care of a while ago now. There is just one function in the public interface to run D2 coloring now:  ``KokkosGraph::Experimental::graph_color_distance2``, so I'll go ahead and close this (almost 8 years later!)
